pkg/client/client.go

```diff
@@ -2055,3 +2023,24 @@ func deleteOptions(dp metav1.DeletionPropagation) metav1.DeleteOptions {
 		PropagationPolicy: &dp,
 	}
 }
+
+// pollUntil is a wrapper around wait.PollUntilContextTimeout that allows adding the passed lastError into
+// the final error if if set by the condition, this would add more context to the "context deadline exceeded" error e.g..
+func pollUntil(ctx context.Context, timeout, interval time.Duration, lastError *error, condition wait.ConditionWithContextFunc) error {
+	if err := wait.PollUntilContextTimeout(ctx, interval, timeout, false, condition); err != nil {
+		if lastError != nil && *lastError != nil {
+			err = fmt.Errorf("%w: lastError: %w", err, *lastError)
+		}
+		return err
+	}
+	return nil
+}
+
+func Poll(ctx context.Context, lastError *error, condition wait.ConditionWithContextFunc) error {
+	return pollUntil(ctx, 5*time.Minute, 3*time.Second, lastError, condition)
+}
+
+// PollUntilDeleted has a larger timeout as it's acceptable for deletion to take longer.
+func PollUntilDeleted(ctx context.Context, lastError *error, condition wait.ConditionWithContextFunc) error {
```

Comment:
I think it better to use the PollUntil with the duration constants for creation, deletion, modification, etc.
The function `Poll` and `PollUntilDeleted` do not have additional instructions than `pollUntil`, while their names suggest they have dedicated actions for deletion or normal operation. It is confusing.

Reply:
We can come up with better names or add more comments.
Passing the same args over and over could be cumbersome and less readable and more prone to inconsistencies.
Poll and PollUntilDeleted  are easier to use and give us more control over the interval/timeout.